### ISSUE TITLE
Add support for searching within embedded hashes

### ISSUE THREAD

**Issue by joshuabowers** [joshuabowers/search_magic#7]
Given the following model definition,

``` ruby
class Foo
  include Mongoid::Document
  include SearchMagic::FullTextSearch
  field :metadata, :type => Hash
  search_on :metadata
end
```

It should be possible to search for specific key/value pairs within the metadata hash field. Syntax would be:

``` ruby
Foo.search_for("metadata:key:value")
```

Additionally, searching for instances of the model which have a key present, regardless of value, could be achieved by:

``` ruby
Foo.search_for("metadata:key?")
```

**Comment by joshuabowers** [comment]
Commit cdbb7fcac88d9bdc0e30556267ae6b5902889bcd should have proper support for the first type of search: the key-value pair.

**Comment by joshuabowers** [comment]
Presence detection available as of cd6fc8c52fb6fad4fab759f3eea0b3e4886e272f.
